Library/Homebrew/extend/enumerable.rbi

```diff
@@ -0,0 +1,17 @@
+# typed: strict
+
+module Enumerable
+  requires_ancestor { Object }
+
+  sig {
+    type_parameters(:key).params(
+      block: T.nilable(T.proc.params(o: Enumerable::Elem).returns(T.type_parameter(:key))),
+    ).returns(T::Hash[T.type_parameter(:key), Enumerable::Elem])
+  }
+  def index_by(&block); end
+end
+
+class Hash
+  sig { returns(T::Hash[Hash::K, Hash::V]) }
```

Comment:
I would like to use a runtime `sig` with `returns(T.self_type)`, but that's a bug: https://github.com/sorbet/sorbet/issues/7586